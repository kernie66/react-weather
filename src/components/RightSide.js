import { List } from 'reactstrap';

export default function RightSide() {
  return (
    <List type="unstyled" className="outline-md text-end">
      <li className="outline-sm border-bottom">Molntäcke</li>
      <li className="pe-1 pb-4">80%</li>
      <li className="outline-sm border-bottom pt-5">Känns som</li>
      <li className="pe-1 pb-4">27&deg;C</li>
      <li className="outline-sm border-bottom pt-4">Vindbyar</li>
      <li className="pe-1">4 m/s</li>
    </List>
  );
};