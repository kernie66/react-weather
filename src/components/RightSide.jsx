import { Divider, Group, Stack, Text } from '@mantine/core';
import { useWeatherData } from '../utils/weatherQueries.js';
import { useEffect, useState } from 'react';

export default function RightSide() {
  const { data: weatherData } = useWeatherData();
  const [windGust, setWindGust] = useState(0);

  useEffect(() => {
    let newWindGust = weatherData?.current.wind_speed;

    if (weatherData) {
      if (weatherData.current.wind_gust) {
        newWindGust = weatherData.current.wind_gust;
      } else if (weatherData.hourly[0].wind_gust) {
        newWindGust = weatherData.hourly[0].wind_gust;
      }
    }

    setWindGust(newWindGust);
  }, [weatherData]);

  return (
    <Stack h="100%" justify="space-around" gap="md">
      <Stack gap={4}>
        <Group justify="flex-end">
          <Text className="outline-sm">Molntäcke</Text>
        </Group>
        <Divider />
        <Group justify="flex-end">
          <Text className="outline-md">
            {weatherData?.current.clouds}%
          </Text>
        </Group>
      </Stack>
      <Stack gap={4}>
        <Group justify="flex-end">
          <Text className="outline-sm">Känns som</Text>
        </Group>
        <Divider />
        <Group justify="flex-end">
          <Text className="outline-md">
            {weatherData?.current.feels_like.toFixed(1)}&deg;C
          </Text>
        </Group>
      </Stack>
      <Stack gap={4}>
        <Group justify="flex-end">
          <Text className="outline-sm">Vindbyar</Text>
        </Group>
        <Divider />
        <Group justify="flex-end">
          <Text className="outline-md">
            {windGust.toFixed(1)} m/s
          </Text>
        </Group>
      </Stack>
    </Stack>
  );
}
