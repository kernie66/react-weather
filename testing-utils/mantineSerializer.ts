// @see https://github.com/thymikee/jest-preset-angular/issues/336#issuecomment-1536232102
// @see https://github.com/orgs/mantinedev/discussions/467
const attributesToClean: { [key: string]: RegExp[] } = {
  class: [/^m-.*$/],
  id: [/^mantine-.*$/],
  for: [/^mantine-.*$/],
  'aria-describedby': [/^mantine-.*$/],
  'aria-labelledby': [/^mantine-.*$/],
  'aria-controls': [/^mantine-.*$/],
};
const attributesToCleanKeys = Object.keys(attributesToClean);
const hasAttributesToClean = (attribute: Attr) =>
  attributesToCleanKeys.some((name) => attribute.name === name);

let lastCleanedNode: Element | null = null;

module.exports = {
  print: (val: Element, serialize: (v: Element) => string) => {
    const clone = val.cloneNode(true) as Element;

    for (const attr of Object.values(clone.attributes).filter(
      hasAttributesToClean
    )) {
      attr.value = attr.value
        .split(' ')
        .filter((attrValue: string) => {
          return !attributesToClean[attr.name].some((regex) =>
            regex.test(attrValue)
          );
        })
        .join(' ');
    }

    lastCleanedNode = clone;
    return serialize(clone);
  },
  test: (val: Element) =>
    val !== lastCleanedNode &&
    val.attributes !== undefined &&
    Object.values(val.attributes).some(hasAttributesToClean),
};
