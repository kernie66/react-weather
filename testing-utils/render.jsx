// ./test-utils/render.tsx
import { render as testingLibraryRender } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
// Import your theme object
import { createTheme } from '@mantine/core';
import {
  QueryClient,
  QueryClientProvider,
} from '@tanstack/react-query';

const theme = createTheme({});

const createTestQueryClient = () =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
    logger: {
      log: console.log,
      warn: console.warn,
      error: () => {},
    },
  });

export const testQueryClient = createTestQueryClient();

export function render(ui) {
  console.log('testQueryClient', testQueryClient);
  return testingLibraryRender(<>{ui}</>, {
    wrapper: ({ children }) => (
      <MantineProvider theme={theme}>
        <QueryClientProvider client={testQueryClient}>
          {children}
        </QueryClientProvider>
      </MantineProvider>
    ),
  });
}
