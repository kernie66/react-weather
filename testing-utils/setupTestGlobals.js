// src/test-globals.ts
export const setup = () => {
  process.env.TZ = 'Europe/Stockholm';
};
