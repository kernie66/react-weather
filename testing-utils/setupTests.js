import { afterAll, beforeAll } from 'vitest';
import { server } from '../src/mocks/server.js';
import dayjs from 'dayjs';
import 'dayjs/locale/sv';
import localizedFormat from 'dayjs/plugin/localizedFormat.js';
import calendar from 'dayjs/plugin/calendar.js';

beforeAll(() => {
  server.listen();

  dayjs.locale('sv');
  dayjs.extend(localizedFormat);
  dayjs.extend(calendar);
});

afterEach(() => server.resetHandlers());

afterAll(() => server.close());
