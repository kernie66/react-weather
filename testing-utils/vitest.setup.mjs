// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom/vitest';

import { vi } from 'vitest';

const { getComputedStyle } = window;
window.getComputedStyle = (elt) => getComputedStyle(elt);
window.HTMLElement.prototype.scrollIntoView = () => {};

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: vi.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: vi.fn(),
    removeListener: vi.fn(),
    addEventListener: vi.fn(),
    removeEventListener: vi.fn(),
    dispatchEvent: vi.fn(),
  })),
});

class ResizeObserver {
  observe() {}
  unobserve() {}
  disconnect() {}
}

window.ResizeObserver = ResizeObserver;
