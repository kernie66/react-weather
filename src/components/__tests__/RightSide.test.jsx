import { render, screen } from '../../../testing-utils';
import weatherDataJson from '../../../bruno/weather_alert_response.json';
import dayjs from 'dayjs';
import { testQueryClient } from '../../../testing-utils/render.jsx';
import { vi, expect } from 'vitest';
import RightSide from '../RightSide.jsx';

const weatherDate = dayjs.unix(weatherDataJson.current.dt);
const fakeTodayDate = new Date(weatherDate.toDate());

describe('test RightSide', () => {
  beforeEach(() => {
    testQueryClient.removeQueries();
    localStorage.clear();
  });

  afterAll(() => {
    vi.useRealTimers();
  });

  it('renders right side info with current wind gust', async () => {
    vi.setSystemTime(fakeTodayDate);
    const snapshot = render(<RightSide />);

    expect(await screen.findByText(/molntäcke/i)).toBeInTheDocument();
    screen.debug();
    expect(screen.getByText(/känns som/i)).toBeInTheDocument();
    expect(screen.getByText(/luftfuktighet/i)).toBeInTheDocument();
    expect(snapshot).toMatchSnapshot();
  });
});
