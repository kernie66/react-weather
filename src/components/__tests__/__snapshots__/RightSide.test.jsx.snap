// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test RightSide > renders right side info with current wind gust 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <style
        data-mantine-styles="classes"
      >
        @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
      </style>
      <div
        class="right-side--stack m_6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: space-between; min-height: 100%;"
      >
        <div
          class="m_6d731127 mantine-Stack-root"
          style="--stack-gap: calc(0.25rem * var(--mantine-scale)); --stack-align: stretch; --stack-justify: flex-start;"
        >
          <div
            class="m_4081bf90 mantine-Group-root"
            style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap;"
          >
            <p
              class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
            >
              Molntäcke
            </p>
          </div>
          <div
            class="m_3eebeb36 mantine-Divider-root"
            data-orientation="horizontal"
            role="separator"
          />
          <div
            class="m_4081bf90 mantine-Group-root"
            style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap;"
          >
            <p
              class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
            >
              44
              %
            </p>
          </div>
        </div>
        <div
          class="m_6d731127 mantine-Stack-root"
          style="--stack-gap: calc(0.25rem * var(--mantine-scale)); --stack-align: stretch; --stack-justify: flex-start;"
        >
          <div
            class="m_4081bf90 mantine-Group-root"
            style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap;"
          >
            <p
              class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
            >
              Känns som
            </p>
          </div>
          <div
            class="m_3eebeb36 mantine-Divider-root"
            data-orientation="horizontal"
            role="separator"
          />
          <div
            class="m_4081bf90 mantine-Group-root"
            style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap;"
          >
            <p
              class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
            >
              21.0
              °C
            </p>
          </div>
        </div>
        <div
          class="m_6d731127 mantine-Stack-root"
          style="--stack-gap: calc(0.25rem * var(--mantine-scale)); --stack-align: stretch; --stack-justify: flex-start;"
        >
          <div
            class="m_4081bf90 mantine-Group-root"
            style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap;"
          >
            <p
              class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
            >
              Vindbyar
            </p>
          </div>
          <div
            class="m_3eebeb36 mantine-Divider-root"
            data-orientation="horizontal"
            role="separator"
          />
          <div
            class="m_4081bf90 mantine-Group-root"
            style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap;"
          >
            <p
              class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
            >
              2.8
               m/s
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <style
      data-mantine-styles="classes"
    >
      @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
    </style>
    <div
      class="right-side--stack m_6d731127 mantine-Stack-root"
      style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: space-between; min-height: 100%;"
    >
      <div
        class="m_6d731127 mantine-Stack-root"
        style="--stack-gap: calc(0.25rem * var(--mantine-scale)); --stack-align: stretch; --stack-justify: flex-start;"
      >
        <div
          class="m_4081bf90 mantine-Group-root"
          style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap;"
        >
          <p
            class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
          >
            Molntäcke
          </p>
        </div>
        <div
          class="m_3eebeb36 mantine-Divider-root"
          data-orientation="horizontal"
          role="separator"
        />
        <div
          class="m_4081bf90 mantine-Group-root"
          style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap;"
        >
          <p
            class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
          >
            44
            %
          </p>
        </div>
      </div>
      <div
        class="m_6d731127 mantine-Stack-root"
        style="--stack-gap: calc(0.25rem * var(--mantine-scale)); --stack-align: stretch; --stack-justify: flex-start;"
      >
        <div
          class="m_4081bf90 mantine-Group-root"
          style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap;"
        >
          <p
            class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
          >
            Känns som
          </p>
        </div>
        <div
          class="m_3eebeb36 mantine-Divider-root"
          data-orientation="horizontal"
          role="separator"
        />
        <div
          class="m_4081bf90 mantine-Group-root"
          style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap;"
        >
          <p
            class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
          >
            21.0
            °C
          </p>
        </div>
      </div>
      <div
        class="m_6d731127 mantine-Stack-root"
        style="--stack-gap: calc(0.25rem * var(--mantine-scale)); --stack-align: stretch; --stack-justify: flex-start;"
      >
        <div
          class="m_4081bf90 mantine-Group-root"
          style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap;"
        >
          <p
            class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
          >
            Vindbyar
          </p>
        </div>
        <div
          class="m_3eebeb36 mantine-Divider-root"
          data-orientation="horizontal"
          role="separator"
        />
        <div
          class="m_4081bf90 mantine-Group-root"
          style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-end; --group-wrap: wrap;"
        >
          <p
            class="mantine-focus-auto _outlineSingle_597f95 m_b6d8b162 mantine-Text-root"
          >
            2.8
             m/s
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
